libs/server/repository/shell/src/lib/factories/pr-repository.factory.ts

```diff
@@ -1,20 +1,20 @@
-import { PmpApiConfigService } from '@pimp-my-pr/server/shared/core';
 import { HttpService } from '@nestjs/common';
+
 import {
   BitbucketPrRepository,
   GithubPrRepository
 } from '@pimp-my-pr/server/repository/infrastructure';
+import { Platform } from '@pimp-my-pr/shared/domain';
 
-export const prRepositoryFactory = (
-  configService: PmpApiConfigService,
-  httpService: HttpService
-) => {
-  if (configService.getBitbucketToken()) {
-    return new BitbucketPrRepository(httpService);
-  }
-  if (configService.getGithubToken()) {
-    return new GithubPrRepository(httpService);
-  }
+export const prRepositoryFactoryFactory = (httpService: HttpService) => (platform: Platform) => {
+  switch (platform) {
+    case Platform.github:
+      return new GithubPrRepository(httpService);
```

Comment:
you can consider also to provide here the remote repositories  instead of httpService for better performence. Notice that in every place where repositories are used you need to trigger constructing a new service. Nestjs recommended using a singleton version of services.